query/src/main/scala/filodb/query/exec/MultiSchemaPartitionsExec.scala

```diff
@@ -49,6 +45,9 @@ final case class MultiSchemaPartitionsExec(id: String,
     val lookupRes = source.lookupPartitions(dataset, partMethod, chunkMethod)
     Kamon.currentSpan().mark("lookup-partitions-done")
 
+    val queryTime = (System.currentTimeMillis() - queryContext.submitTime) / 1000
+    if (queryTime >= queryContext.queryTimeoutSecs) throw QueryTimeoutException(queryTime, this.getClass.getName)
```

Comment:
As already suggested, queryTimeout should be in millis so it is more accurate, and to also avoid the divide by 1000 everywhere.